xbmc/platform/android/activity/XBMCApp.cpp

```diff
@@ -1251,13 +1251,13 @@ int CXBMCApp::WaitForActivityResult(const CJNIIntent &intent, int requestCode, C
     ret = event->GetResultCode();
   }
 
-  // delete from m_activityResultEvents map before deleting the event
+  // delete from m_activityResultEvents before deleting the event
   CSingleLock lock(m_activityResultMutex);
-  for (auto it = m_activityResultEvents.begin(); it != m_activityResultEvents.end(); ++it)
+  for (const auto& activityResultEvent : m_activityResultEvents)
   {
-    if ((*it)->GetRequestCode() == requestCode)
+    if (activityResultEvent->GetRequestCode() == requestCode)
     {
-      m_activityResultEvents.erase(it);
+      m_activityResultEvents.erase(activityResultEvent);
```

Comment:
Hmm, isn't erase with a key_type way slower than erase with an iterator?

https://www.cplusplus.com/reference/map/map/erase/

Complexity
For the first version (erase(position)), amortized constant.
For the second version (erase(val)), logarithmic in container size.

Reply:
mh, i assumed we get an iterator returned by the range based loop, now looking at it again i'm figuring if an erase with a const ref is possible at all... is that runtime tested, @enen92 ?
sorry for any confusion

Reply:
Not runtime tested, building in jenkins as explained in the pr description

Reply:
That will build, but as i said, it has runtime performance implications.

Reply:
@ksooo I also agree the style changes are not worth a penalty on performance (although the vector is/will always be small). Conclusion: close this one and reopen the original backport?